From a256ef56922bdcb06e1144f95aca010c3916482f Mon Sep 17 00:00:00 2001
From: Anuj Aggarwal <anuj.aggarwal@ti.com>
Date: Fri, 12 Jun 2009 18:16:34 +0530
Subject: [PATCH 426/426] smsc911x ethernet driver : save MAC address before soft reset (2)

This patch is required, on top of the previous commit (cfebe15),
to properly save and restore the MAC address.
---
 drivers/net/smsc911x.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/drivers/net/smsc911x.c b/drivers/net/smsc911x.c
index ee7db39..9478adc 100644
--- a/drivers/net/smsc911x.c
+++ b/drivers/net/smsc911x.c
@@ -1835,7 +1835,7 @@ static int __devinit smsc911x_init(struct net_device *dev)
 		SMSC_WARNING(PROBE,
 			"This driver is not intended for this chip revision");
 
-        /* save MAC address */
+	/* save MAC address */
 	 mac_high16 = smsc911x_mac_read(pdata, ADDRH);
 	 mac_low32 = smsc911x_mac_read(pdata, ADDRL);
 
@@ -1847,6 +1847,14 @@ static int __devinit smsc911x_init(struct net_device *dev)
 	smsc911x_mac_write(pdata, ADDRH, mac_high16);
 	smsc911x_mac_write(pdata, ADDRL, mac_low32);
 
+	/* write it to the dev structure as well */
+	dev->dev_addr[0] = (u8)(mac_low32);
+	dev->dev_addr[1] = (u8)(mac_low32 >> 8);
+	dev->dev_addr[2] = (u8)(mac_low32 >> 16);
+	dev->dev_addr[3] = (u8)(mac_low32 >> 24);
+	dev->dev_addr[4] = (u8)(mac_high16);
+	dev->dev_addr[5] = (u8)(mac_high16 >> 8);
+
 	/* Disable all interrupt sources until we bring the device up */
 	smsc911x_reg_write(pdata, INT_EN, 0);
 
-- 
1.6.2.4

